how-to-guides/custom-events-pass.ipynb

```diff
@@ -0,0 +1,90 @@
+{
+ "cells": [
+  {
+   "cell_type": "markdown",
+   "source": [
+    "# How to create Custom Events based on Forward or Backward Pass"
+   ],
+   "metadata": {}
+  },
+  {
+   "cell_type": "markdown",
+   "source": [
+    "Ignite provides you the flexibility to add events based on the loss\n",
+    "calculation and backward pass as follows:"
+   ],
+   "metadata": {}
+  },
+  {
+   "cell_type": "code",
+   "execution_count": 1,
+   "source": [
+    "from ignite.engine import EventEnum, Engine\n",
+    "\n",
+    "class BackpropEvents(EventEnum):\n",
+    "    \"\"\"\n",
+    "    Events based on backpropagation\n",
+    "    \"\"\"\n",
+    "    BACKWARD_STARTED = 'backward_started'\n",
+    "    BACKWARD_COMPLETED = 'backward_completed'\n",
+    "    OPTIM_STEP_COMPLETED = 'optim_step_completed'\n",
+    "\n",
+    "def update(engine, batch):\n",
+    "    model.train()\n",
+    "    opitmizer.zero_grad()\n",
+    "    x, y = process_batch(batch)\n",
+    "    y_pred = model(x)\n",
```

Comment:
Let's make this code snippet self-contained and executable

Reply:
Agree! That's why we use notebooks for this